pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -483,17 +497,20 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
      * @param kernel Kernel to run the operation
      * @param matrix Pointer to the array data (in row-major format).
      * @param controlled_wires Control wires.
+     * @param controlled_values Control values (0 or 1).
      * @param wires Wires to apply gate to.
      * @param inverse Indicate whether inverse should be taken.
      */
     inline void applyControlledMatrix(
         const ComplexT *matrix, const std::vector<size_t> &controlled_wires,
+        const std::vector<bool> &controlled_values,
         const std::vector<size_t> &wires, bool inverse = false) {
         const auto &dispatcher = DynamicDispatcher<PrecisionT>::getInstance();
         auto *arr = this->getData();
-
         PL_ABORT_IF(wires.empty(), "Number of wires must be larger than 0");
-
+        PL_ABORT_IF_NOT(controlled_wires.size() == controlled_values.size(),
```

Comment:
Do we want to also make sure `wires.size()>=controlled_wires.size()`?

Reply:
No, these are independent. We could check that they are disjoint sets, but I think this is better done at the Python layer.